asf_tools/composite.py

```diff
@@ -12,29 +12,48 @@
 import glob
 import logging
 import os
+import statistics
+from typing import List
 
 import numpy as np
-from hyp3lib import saa_func_lib as saa
 from osgeo import gdal, osr
 
 
-def get_epsg_code(file_name):
-    info = gdal.Info(file_name, format='json')
+def get_epsg_code(info: dict) -> int:
+    """Get the EPSG code from a GDAL Info dictionary"""
     proj = osr.SpatialReference(info['coordinateSystem']['wkt'])
-    epsg_code = proj.GetAttrValue('AUTHORITY', 1)
-    return f'EPSG:{epsg_code}'
+    epsg_code = int(proj.GetAttrValue('AUTHORITY', 1))
+    return epsg_code
 
 
-def get_target_epsg_code(files):
-    epsg_codes = [get_epsg_code(f) for f in files]
+def get_target_epsg_code(codes: List[int]) -> int:
+    """Determine the target UTM EPSG projection for the output mosaic
 
+    Args:
+        codes: List of UTM EPSG codes
+
+    Returns:
+        target: UTM EPSG code
+    """
     # use median east/west UTM zone of all files, regardless of hemisphere
-    zones = [int(epsg_code[-2:]) for epsg_code in epsg_codes]
-    target_zone = int(np.median(zones))
+    # UTM EPSG codes for each hemisphere will look like:
+    #   North: 326XX
+    #   South: 327XX
+    code_array = np.array(codes)
+    valid_codes = np.concatenate([np.arange(32601, 32661), np.arange(32701, 32761)])
+    if not np.isin(code_array, valid_codes).all():
+        raise ValueError(f'Non UTM EPSG code encountered: {codes}')
+    hemispheres = code_array // 100 * 100
+    zones = code_array % 100
+
+    # handle antimeridian
+    target_zone = int(statistics.median(zones % 60))
+    if target_zone == 0:
+        target_zone = 60
```

Comment:
This possibly just moves the problem from 60 --> 01 to 59 --> 0

Might have to check `0 in zones and 60 in zones` (or that there is scenes on both sides...) and then shift by 30, calculate the median and shift back

Reply:
possibly use the length of the set of zones to determine shift...

Reply:
What is the problem you are trying to fix?  I don't see a problem at the antimeridian.  Median should grab the value that appears MOST in an array, correct?  (that is why I used median and not average)  Why is zones % 60 needed at all?  Zones don't go past 59, do they?

Reply:
[median](https://docs.python.org/3/library/statistics.html#statistics.median) grabs the middle value, after sorting, for an *odd* numbered array; for an *even* numbered array, it takes the average of the two middle values. So, since UTM zones are numbered 1-60, a composite evenly split across the antimerridian will give:
```python
>>> from statistics import median
>>> zones = [60, 1]
>>> int(median(zones))
30
```

Reply:
The most robust way to do this would be to sum unit vectors for each scene's zone, and then determine the correct zone from the resulting vector. 

This method (or extending to the principle axis assuming some [underlying spherical distribution](https://en.wikipedia.org/wiki/Directional_statistics)) would also handle latitude bands if we go to MGRS.